playground/streaming/benchmark.py

```diff
@@ -0,0 +1,551 @@
+
+reader = InMemoryMetricReader()
+provider = MeterProvider(metric_readers=[reader])
+metrics.set_meter_provider(provider)
+
+
+async def run_agents():
+    for agent_name in tqdm(args.agents, desc="Agents"):
+        for agent_run_idx in tqdm(range(args.agent_num_cycles), desc="Agent Cycles"):
+            company, model_name = agent_name.split("_")
+
+            length_meter = meter.create_counter(
+                f"agent.agent_chat_{company}-{model_name}.total_characters",
+            )
+
+            if company == "gpt":
+                agent = ChatGPTAgent(
+                    ChatGPTAgentConfig(
+                        initial_message=None,
+                        prompt_preamble=args.agent_prompt_preamble,
+                        allow_agent_to_be_cut_off=False,
+                        model_name=model_name,
+                    )
+                )
+            elif company == "anthropic":
+                agent = ChatAnthropicAgent(
+                    ChatAnthropicAgentConfig(
+                        initial_message=None,
+                        allow_agent_to_be_cut_off=False,
+                        model_name=model_name,
+                    )
+                )
+            agent.attach_transcript(Transcript())
+            agent_task = agent.start()
+            message = AgentInput(
+                transcription=Transcription(
+                    message=args.agent_first_input, confidence=1.0, is_final=True
+                ),
+                conversation_id=0,
+            )
+            await agent.input_queue.put(InterruptibleEvent(message))
+
+            while True:
+                try:
+                    message = await asyncio.wait_for(
+                        agent.output_queue.get(), timeout=15
+                    )
+                    length_meter.add(len(message.payload.message.text))
+                    logger.debug(
+                        f"[Agent: {agent_name}] Response from API: {message.payload.message.text}"
+                    )
+                except asyncio.TimeoutError:
+                    logger.debug(
+                        f"[Agent: {agent_name}] Agent queue is empty, stopping..."
+                    )
+                    break
+
+
+async def run_synthesizers():
+    def create_file_output_device(synthesizer_name, extra_info=""):
+        return FileOutputDevice(
+            os.path.join(args.results_dir, f"{synthesizer_name}{extra_info}.wav"),
+        )
+
+    for synthesizer_cycle_idx in tqdm(
+        range(args.synthesizer_num_cycles), desc="Synthesizer Cycles"
+    ):
+        for synthesizer_name in args.synthesizers:
+            file_output = create_file_output_device(
+                synthesizer_name, f"-run={synthesizer_cycle_idx}"
+            )
+            synthesizer_class, synthesizer_config_class = synthesizer_classes[
+                synthesizer_name
+            ]
+            extra_config = {}
+            if synthesizer_name == "playht":
```

Comment:
should we add these directly as defaults to the synthesizers?

Reply:
Probably. That would make it easier to get started using them.